cli/tests/test_resolve_command.rs

```diff
@@ -552,14 +552,14 @@ fn test_simplify_conflict_sides() {
     >>>>>>> Conflict 1 of 1 ends
     "###);
 
-    // TODO: The conflict should be simplified before being resolved.
-    let error = test_env.jj_cmd_failure(&repo_path, &["resolve"]);
-    insta::assert_snapshot!(error, @r###"
-    Hint: Using default editor ':builtin'; run `jj config set --user ui.merge-editor :builtin` to disable this message.
-    Resolving conflicts in: file1
-    Error: Failed to resolve conflicts
-    Caused by: The conflict at "file1" has 3 sides. At most 2 sides are supported.
-    "###);
+    check_resolve_produces_input_file(&mut test_env, &repo_path, "file1", "base", "base\n");
+    check_resolve_produces_input_file(&mut test_env, &repo_path, "file1", "left", "a\n");
+    check_resolve_produces_input_file(&mut test_env, &repo_path, "file1", "right", "b\n");
+    // Manually resolve conflict in file 1.
+    std::fs::write(repo_path.join("file1"), "resolved").unwrap();
+    check_resolve_produces_input_file(&mut test_env, &repo_path, "file2", "base", "base\n");
+    check_resolve_produces_input_file(&mut test_env, &repo_path, "file2", "left", "a\n");
+    check_resolve_produces_input_file(&mut test_env, &repo_path, "file2", "right", "b\n");
```

Comment:
Does it also work if we edit the conflict without simplifying it? Even without https://github.com/martinvonz/jj/blob/d31edd4ac5092032333662c53264d590535ab415/lib/src/merged_tree.rs#L1250 ?

Reply:
If we don't simplify, `jj resolve` doesn't currently support >2-sided merges so we can't edit here. Perhaps you're referring to some other test?

Reply:
Oops, I meant "without *resolving* it". Sorry!

Reply:
Updated to add a test showing that `resolve` can still preserve the conflict.